src/track/track.h

```diff
@@ -262,21 +262,45 @@ class Track : public QObject {
     ConstWaveformPointer getWaveformSummary() const;
     void setWaveformSummary(ConstWaveformPointer pWaveform);
 
-    // Get the track's main cue point
-    CuePosition getCuePoint() const;
+    /// Get the track's main cue point
+    mixxx::audio::FramePos getMainCuePosition() const;
+    CuePosition getCuePoint() const {
+        const auto position = getMainCuePosition();
+        if (!position.isValid()) {
+            return {};
+        }
+        return {position.toEngineSamplePos()};
```

Comment:
Why do you use fromEngineSamplePosMaybeInvalid() (see below) but not toEngineSamplePosMaybeInvalid() here?

Reply:
902e4a8a8d44ada9009e8b08cedce0ade04cbec4